features/FEATURE_BLE/ble/pal/Deprecated.h

```diff
@@ -19,27 +19,20 @@
 /**
  * Declare the beginning of a code block that uses a deprecated API
  */
-#if defined(__GNUC__) && !defined(__CC_ARM)
+#if defined(__GNUC__)
 #define BLE_DEPRECATED_API_USE_BEGIN() \
     _Pragma("GCC diagnostic push") \
     _Pragma("GCC diagnostic ignored \"-Wdeprecated-declarations\"")
-#elif defined(__CC_ARM)
-#define BLE_DEPRECATED_API_USE_BEGIN() \
-    _Pragma("push") \
-    _Pragma("diag_suppress 1361")
```

Comment:
Could you use the clang pragma instead when the file is compiled with ARMC6 ? 
Please have a look at: http://infocenter.arm.com/help/index.jsp?topic=/com.arm.doc.dui0742b/chr1398848167694.html